GetRCConfiguration/__tests__/handler.test.ts

```diff
@@ -31,27 +41,50 @@ describe("handleEmptyErrorResponse ", () => {
 });
 
 describe("getRCConfigurationHandler", () => {
-  test("should return an IResponseSuccessJson if the model return a valid configuration and the userId match", async () => {
+  test("should return an IResponseSuccessJson calling the model if redis does not return a valid configuration and the userId match", async () => {
+    getTaskMock.mockReturnValueOnce(TE.right(O.none));
     findLastVersionMock.mockReturnValueOnce(
       TE.right(O.some(aRemoteContentConfiguration))
     );
     const r = await getRCConfigurationHandler({
       rccModel: rccModelMock,
-      config: aConfig
+      config: aConfig,
+      redisClient: redisClientMock
+    })({
+      configurationId: aRemoteContentConfiguration.configurationId,
+      userId: aRemoteContentConfiguration.userId
+    });
+    expect(r.kind).toBe("IResponseSuccessJson");
+    expect(getTaskMock).toHaveBeenCalled();
+    expect(findLastVersionMock).toHaveBeenCalled();
+  });
+
+  test("should return an IResponseSuccessJson without calling the model if redis return a valid configuration and the userId match", async () => {
+    getTaskMock.mockReturnValueOnce(
+      TE.right(O.some(JSON.stringify(aRetrievedRCConfiguration)))
+    );
+    const r = await getRCConfigurationHandler({
+      rccModel: rccModelMock,
+      config: aConfig,
+      redisClient: redisClientMock
     })({
       configurationId: aRemoteContentConfiguration.configurationId,
       userId: aRemoteContentConfiguration.userId
     });
     expect(r.kind).toBe("IResponseSuccessJson");
+    expect(getTaskMock).toHaveBeenCalled();
+    expect(findLastVersionMock).not.toHaveBeenCalled();
   });
 
   test("should return an IResponseSuccessJson if the model return a valid configuration and the user-id is internal", async () => {
+    getTaskMock.mockReturnValueOnce(TE.right(O.none));
```

Comment:
for every test we should expect that `getTaskMock` is called, and if this is mocked to return `none` then we should expect also that a `setTaskMock` will be called.

Reply:
[4ee62e6](https://github.com/pagopa/io-functions-service-messages/pull/57/commits/4ee62e697113bb7c183363c2c586eec3a68ca1f7)